src/esp/gfx/PbrShader.cpp

```diff
@@ -318,9 +374,17 @@ PbrShader& PbrShader::setObjectId(unsigned int objectId) {
   return *this;
 }
 
+PbrShader& PbrShader::setPrefilteredMapMipLevels(unsigned int mipLevels) {
+  CORRADE_ASSERT(flags_ & Flag::ImageBasedLighting,
+                 "PbrShader::setPrefilteredMapMipLevels(): the shader was not "
+                 "created with image based lighting enabled",
+                 *this);
+  setUniform(prefilteredMapMipLevelsUniform_, mipLevels);
+  return *this;
+}
+
 PbrShader& PbrShader::setBaseColor(const Mn::Color4& color) {
-  // TODO: remove this constraint after IBL is introduced.
-  if (lightCount_ != 0u) {
+  if (lightCount_ != 0u || flags_ & Flag::ImageBasedLighting) {
```

Comment:
This snippit s repeated a lot, maybe you should factor it out into a method that returns a bool